src/scala/com/github/johnynek/bazel_deps/MakeDeps.scala

```diff
@@ -92,24 +92,50 @@ object MakeDeps {
             System.exit(-1)
             sys.error("exited already")
         }
-        // Build up the IO operations that need to run. Till now,
-        // nothing was written
-        val io = for {
-          _ <- IO.recursiveRmF(IO.Path(model.getOptions.getThirdPartyDirectory.parts))
-          wsp = IO.path(workspacePath)
-          _ <- IO.mkdirs(wsp.parent)
-          _ <- IO.writeUtf8(wsp, ws)
-          _ <- IO.writeUtf8(wsp.sibling("BUILD"), "")
-          builds <- Writer.createBuildFiles(model.getOptions.getBuildHeader, targets)
-        } yield builds
 
-        // Here we actually run the whole thing
-        io.foldMap(IO.fileSystemExec(projectRoot.toFile)) match {
-          case Failure(err) =>
-            System.err.println(err)
-            System.exit(-1)
-          case Success(builds) =>
-            println(s"wrote ${targets.size} targets in $builds BUILD files")
+        if (g.checkOnly) {
+          // Build up the IO operations that need to run.
+          val wsp = IO.path(workspacePath)
+          val io = for {
+            wsOK <- IO.compare(wsp, ws)
+            wsbOK <- IO.compare(wsp.sibling("BUILD"), "")
+            buildsOK <- Writer.compareBuildFiles(model.getOptions.getBuildHeader, targets)
+          } yield wsOK :: wsbOK :: buildsOK
+
+          // Here we actually run the whole thing
+          io.foldMap(IO.fileSystemExec(projectRoot.toFile)) match {
+            case Failure(err) =>
+              System.err.println(err)
+              System.exit(-1)
+            case Success(comparisons) =>
+              val mismatchedFiles = comparisons.filter(!_.ok)
+              if (mismatchedFiles.isEmpty) {
+                System.err.println("all generated files are up-to-date")
```

Comment:
why `System.err` here? Do you want to include the any statistics as we do below (the total number of targets, I guess... not sure how useful it is, handy to see if numbers look totally crazy sometimes).

Reply:
I was following bazel's pattern of putting all diagnostic information on stderr.  I can change this one to stdout if you prefer.  I'm happy without the statistics, personally, but I can add them if you'd like them.  What do you say?

Reply:
I don't really care about err vs out, I just want to be consistent in the two branches. Happy to change the other if you prefer. I'll leave it to you.

Reply:
made success messages go to stdout, errors to stderr